libcst/matchers/_matcher_base.py

```diff
@@ -789,26 +869,62 @@ def _attribute_matches(  # noqa: C901
     # types were ignored.
     return _matches(
         cast(Union[MaybeSentinel, RemovalSentinel, libcst.CSTNode], node),
-        cast(Union[BaseMatcherNode, MatchIfTrue], matcher),
+        cast(Union[BaseMatcherNode, MatchIfTrue, MatchMetadata], matcher),
+        metadata_lookup,
     )
 
 
+def _metadata_matches(
+    node: libcst.CSTNode,
+    metadata: Union[
+        MatchMetadata,
+        AllOf[MatchMetadata],
+        OneOf[MatchMetadata],
+        InverseOf[MatchMetadata],
+    ],
+    metadata_lookup: Callable[[meta.ProviderT, libcst.CSTNode], object],
+) -> bool:
+    # Now, evaluate the matcher node itself.
```

Comment:
This comment seems out of place? Inherited from an older version of the code?

Reply:
I think so. Let me rebase and fix.